### DISCUSSION TITLE
Does this work with both HA-scenes and MQTT-scenes?

### DISCUSSION THREAD

**Discussion by JohnBull-1975** [Dirnei/node-red-contrib-zigbee2mqtt-devices#131] [Q&A]
New to this and planning to buy some IKEA Trådlös bulbs and Styrbar (E2001-2002) remotes and wants to cycle through a few pre-defined scenes with click-left and click-right and maybe also cosy-yellow and bright-white by click-up when already on....
Read about scenes and that MQTT scenes should be faster/better for ZigBee devices.
So, before buying devices and starting to test and try to understand - will this work with MQTT-scenes or only HA-scenes?

I also read a discussion about adding support for the Styrbar E2001/2002 remote from last year. Will that function be implemented soon?

**Comment by Dirnei** [comment]
hi, sorry for my late answer :) the scene node is completely independent and was an idea of my own. I thought about adding support for the mqtt scenes, but I haven't had a need for it. Support Styrbar is possible, but I would need the device to test it.

All what you planning to do should be easy in the current version :)

**Reply by JohnBull-1975** [reply]
Hi and thank's for (late) reply :-)
I figured it out after a while, that it was just as you wrote, completely independent, and that Scene-in was a start for a flow, that is built in NR and chosen/altered/selected by Scene-selector. It was not obvious from the documentation though...

A picture of the Scene-in node with some example action on this page: https://github.com/Dirnei/node-red-contrib-zigbee2mqtt-devices/blob/master/docs/nodes/scene-in.md could be helpful for others to understand what it really does, as I at least first thought it was coupled to a HA or MQTT scene. Maybe as simple as this as an example:
![image](https://github.com/Dirnei/node-red-contrib-zigbee2mqtt-devices/assets/121295714/c7e7cefe-5f81-4f71-856f-9ef8841ee12b)

Scene-selector has some pictures, but that doesn't tell how Scene-in works as the Scene-in-nodes there is only connected to a debug node.

And you're right - what I planned was possible and it was fun figuring out how to configure and play around with. After that I ended up with another solution as I returned the RGB lamps as they were not bright enough and I replaced them by some more powerful and dimmable and then I had no real need for scenes any more. I could then pass one action to each button/direction of the Styrbar wall switch.

**Comment by Dirnei** [comment]
Good to hear that you were able to progress :)

I opened #132 to improve the documentation, as you are totally right that it looks like that it would support the scene feature in mqtt.
